Library/Homebrew/os/mac/ruby_mach.rb

```diff
@@ -0,0 +1,101 @@
+require "vendor/macho/macho"
+require "os/mac/architecture_list"
+
+module RubyMachO
+  # @private
+  def macho
+    @macho ||= begin
+      MachO.open(to_s)
+    end
+  end
+
+  # @private
+  def mach_data
+    @mach_data ||= begin
+      machos = []
+      mach_data = []
+
+      if MachO.fat_magic?(macho.magic)
+        machos = macho.machos
+      else
+        machos << macho
+      end
+
+      machos.each do |m|
+        arch = case m.cputype
+        when "CPU_TYPE_X86" then :i386
+        when "CPU_TYPE_X86_64" then :x86_64
+        when "CPU_TYPE_POWERPC" then :ppc7400
+        when "CPU_TYPE_POWERPC64" then :ppc64
+        end
```

Comment:
I would suggest to insert a line `else :dunno` before this one (like in `cctools_mach.rb`) to avoid returning `nil` instead of a symbol, as that will make breakage due to new/unknown architectures easier to spot.